src/xrpld/app/misc/detail/LendingHelpers.cpp

```diff
@@ -0,0 +1,118 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of rippled: https://github.com/ripple/rippled
+    Copyright (c) 2025 Ripple Labs Inc.
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#include <xrpld/app/misc/LendingHelpers.h>
+//
+#include <xrpld/app/tx/detail/Transactor.h>
+#include <xrpld/app/tx/detail/VaultCreate.h>
+
+namespace ripple {
+
+bool
+lendingProtocolEnabled(PreflightContext const& ctx)
+{
+    return ctx.rules.enabled(featureLendingProtocol) &&
+        VaultCreate::isEnabled(ctx);
+}
+
+namespace detail {
+
+Number
+loanPeriodicRate(TenthBips32 interestRate, std::uint32_t paymentInterval)
+{
+    // Need floating point math for this one, since we're dividing by some
+    // large numbers
+    return tenthBipsOfValue(Number(paymentInterval), interestRate) /
+        (365 * 24 * 60 * 60);
+}
+
+Number
```

Comment:
Are there corresponding sections or formulas in the specifications for all of the functions here and in *cpp? It's not easy to verify without a reference. Comments pointing to a location in the specifications would be very helpful.

Reply:
> Are there corresponding sections or formulas in the specifications for all of the functions here and in *cpp? It's not easy to verify without a reference. Comments pointing to a location in the specifications would be very helpful.

Will fix.